pkg/apis/operator/v1/types.go

```diff
@@ -312,6 +316,47 @@ type IPPool struct {
 	BlockSize *int32 `json:"blockSize,omitempty"`
 }
 
+// CNIPluginType describe the type of CNI plugin used.
+type CNIPluginType string
+
+const (
+	PluginCalico    CNIPluginType = "Calico"
+	PluginGKE       CNIPluginType = "GKE"
+	PluginAmazonVPC CNIPluginType = "AmazonVPC"
+	PluginAzureVNET CNIPluginType = "AzureVNET"
+)
+
+var CNIPluginTypes []CNIPluginType = []CNIPluginType{
```

Comment:
Is this used anywhere?

Reply:
I'm not sure if it is. But perhaps I should use it to validate the PluginType.